app/test/qgb_rpc_test.go

```diff
@@ -0,0 +1,98 @@
+package app_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/testutil/testnode"
+	"github.com/celestiaorg/celestia-app/x/qgb/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestQGBRPCQueries(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping QGB integration test in short mode.")
+	}
+	_, cctx := testnode.DefaultNetwork(t, time.Millisecond)
+	h, err := cctx.WaitForHeightWithTimeout(401, time.Minute)
```

Comment:
[definitely optional]
if we don't want to wait a minute here, we can change the genesis state to include a significantly shorter data window param

Reply:
We won't wait a minute. It's just a worst case scenario. Generally, with a block time of 1ms, 401 blocks will be created in less than half a second.
Yes, we can do that, but ,with fast blocktime, i dont see why tinker with the genesis state and do more changes.

Im okey with any tbh, whichever you prefer :cat:

Reply:
its fine as is, just an optimization :+1:

Reply:
block times will likely take longer than a millisecond, just fyi, that's kinda deceiving, as its really just the timeout commit